backend/src/apiserver/server/pipeline_server.go

```diff
@@ -164,13 +189,42 @@ func (s *PipelineServer) ListPipelines(ctx context.Context, request *api.ListPip
 		listPipelineRequests.Inc()
 	}
 
+	filterContext, err := ValidateFilter(request.ResourceReferenceKey)
+	if err != nil {
+		return nil, util.Wrap(err, "Validating filter failed.")
+	}
+
+	refKey := filterContext.ReferenceKey
+	if common.IsMultiUserMode() {
+		if refKey == nil || refKey.Type != common.Namespace {
+			return nil, util.NewInvalidInputError("Invalid resource references for pipelines. ListPipelineVersions requires filtering by namespace.")
+		}
+		namespace := refKey.ID
+		resourceAttributes := &authorizationv1.ResourceAttributes{
+			Namespace: namespace,
+			Verb:      common.RbacResourceVerbList,
+		}
+		err = s.CanAccessPipeline(ctx, "", resourceAttributes)
```

Comment:
We can write if err := s.CanAccessPipeline(ctx, "", resourceAttributes); err != nil {xxx}